### PR TITLE
First pass at #453 (text-based syntaxes should be designed for humans)

### PR DESCRIPTION
**Pull request by LeaVerou**
***
<a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/pull/472.html" title="Last updated on Dec 3, 2024, 4:35 PM UTC (8d2f183)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/472/c28ea64...8d2f183.html" title="Last updated on Dec 3, 2024, 4:35 PM UTC (8d2f183)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -841,6 +841,31 @@ that authors can use to extend the language
...
+Historically, every syntax that could be read by humans, inevitably ended up being authored by humans too.
+Even when syntax is primarily authored by tools,
+it is still effectively authored by humans: the tool developers.
+Therefore, optimizing it for human usability also makes it easier to develop tools to generate it.
+
+Human usability may often be at odds with parsing performance and filesize efficiency.
+Ideally, the syntax would be optimized for human usability, and tooling would optimize the syntax for efficiency.
+However, this is not always possible.
+In these cases, explore if it is possible to provide syntax that is flexible enough to accommodate both use cases.
+The downside of this approach is that it increases the complexity of the language, and humans may still need to read the less human-friendly syntax.
```
Comment (martinthomson) [inline index.bs:858]: I'm not sure what this sentence is getting at.  What less-friendly syntax are you talking about?

Comment (LeaVerou) [reply index.bs:858]: It's talking about cases where you need two have two syntaxes, one optimized for efficiency and one for humans.

File: index.bs [hunk 2]
```diff
@@ -841,6 +841,31 @@ that authors can use to extend the language
...
+Lastly, if efficiency is truly critical, consider whether a binary format may be more appropriate.
+
+<div class="example">
```
Comment (matatk) [inline index.bs:861]: @plinss and @LeaVerou also gave JSON as a great example (+1 to both of these from me):

* Not allowing trailing commas is not good for humans.
* Not allowing comments is *really* not good for humans.

File: index.bs [hunk 3]
```diff
@@ -841,6 +841,31 @@ that authors can use to extend the language
...
+When designing a text-based syntax, consider human usability and readability,
+even when you expect the syntax to be primarily generated by tooling.
+
+Historically, every syntax that could be read by humans, inevitably ended up being authored by humans too.
+Even when syntax is primarily authored by tools,
+it is still effectively authored by humans: the tool developers.
+Therefore, optimizing it for human usability also makes it easier to develop tools to generate it.
+
+Human usability may often be at odds with parsing performance and filesize efficiency.
+Ideally, the syntax would be optimized for human usability, and tooling would optimize the syntax for efficiency.
```
Comment (matatk) [inline index.bs:855]: I was a little unclear as to what this line was trying to say; @LeaVerou explained it's about situations e.g. where you can minify a file with tooling to make it more efficient - that was good clarification I think, so I suggest including that in the text.

It feels like there could be quite big implications here (accepting tooling as integral to the process, if we've not already explicitly done so elsewhere). Solving for both human readability and efficiency is an important challenge.

**Comment by plinss** [comment]
Superseded by #505

**Pull request closed without merging**